swap_meet/vendor.py

```diff
@@ -1,2 +1,76 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        inventory = inventory if inventory is not None else []
+        self.inventory = inventory
+
+    def add(self, new_item):
+        self.inventory.append(new_item)
+        return new_item
+
+    def remove(self, remove_item):
+        if remove_item in self.inventory:
+            self.inventory.remove(remove_item)
+            return remove_item
+        else:
+            return False
+    def get_by_category(self, category):
+        items = []
+        for item in self.inventory:
+            if item.category == category:
+                items.append(item)
+        return items
+    def swap_items(self, vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in vendor.inventory:
+            return False
+        vendor.add(my_item)
+        self.remove(my_item)
+        self.add(their_item)
+        vendor.remove(their_item)
+        return True
+
+    def swap_first_item(self, vendor):
+        if not self.inventory or not vendor.inventory:
+            return False
+        their_first_item = vendor.inventory[0]
+        my_first_item = self.inventory[0]
+        first_item_swap = self.swap_items(vendor, my_first_item, their_first_item)
+        return first_item_swap
```

Comment:
first_item_swap is a boolean value returned from swap_items. Consider renaming the variable to something more descriptive like `successfully_swapped` or `first_item_swapped`. However, you can directly return the value too. 

```python
return self.swap_items(vendor, my_first_item, their_first_item)
```

A quick note about time complexity - It is great to reuse methods we've already written, however swap_items has linear time complexity because it invokes the remove() method. 

Can you think of how you could write a custom swapping implementation to bring the time complexity to O(1) ? Recall that Python has syntax for reassigning values like `a, b = b, a`